fix: filterSpecsFromCoverage should only filter specs

skimi opened the pull request:
#326

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/code-coverage?pullRequest=327) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/code-coverage?pullRequest=327) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/code-coverage?pullRequest=327) it.</sub>

bahmutov reviewed (approved):
@skimi overall I love it, makes sense. Could you
1. resolve the merge conflicts, since some logic now clashes
2. set up an example folded following other examples to verify that it really works and removes just the spec name and not the source file? Or some kind of test for this logic

bahmutov reviewed (changes requested):
Ooops accidentally clicked "Approve", meant to say "Request changes"

skimi commented:
@bahmutov I've rebased to master an I've added a test for the reported case

But to be honest, imo the function could just filter out every files in the integration folder. I don't really see a case where one would want the code coverage to include anything from the integration folder 🤷‍♂️. So all the logic looking at the `testFiles` pattern could just be removed imo
